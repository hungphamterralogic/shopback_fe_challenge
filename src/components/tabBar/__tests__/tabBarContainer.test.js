import React from 'react';
import { genInitialMockStore, renderWithStore } from 'test/reduxComponentTestHelpers';
import initialMockReduxStore from 'test/mockReduxStore/initialState.json';
import dataPopulatedMockReduxStore from 'test/mockReduxStore/dataPopulatedState.json';

import TabBarContainer from '../tabBarContainer';

describe('TabBarContainer', () => {
  test('not yet populated redux store', () => {
    const mockStore = genInitialMockStore(initialMockReduxStore);
    const renderer = renderWithStore(<TabBarContainer />, mockStore);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  test('populated redux store', () => {
    const mockStore = genInitialMockStore(dataPopulatedMockReduxStore);
    const renderer = renderWithStore(<TabBarContainer />, mockStore);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
