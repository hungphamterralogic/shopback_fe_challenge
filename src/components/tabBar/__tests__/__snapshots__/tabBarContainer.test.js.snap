// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarContainer not yet populated redux store 1`] = `
<div
  className="TabBar-tabBar-0-1-1"
/>
`;

exports[`TabBarContainer populated redux store 1`] = `
<div
  className="TabBar-tabBar-0-1-1"
>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5 TabItem-selected-0-1-7"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/tshirt.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Fashion
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/fire.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Popular
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/burger.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Dining
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/plane.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Travel
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/laptop.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Electronics
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/lettuce.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Groceries
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/lipstick.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Health & Beauty
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/couch.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Home & Living
    </span>
  </div>
  <div
    className="TabItem-tabItem-0-1-4 TabItem-tabItemDestop-0-1-5"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="TabItem-icon-0-1-8"
      src="https://cloud.shopback.com/raw/upload/static/images/icon/core/basketball.svg"
    />
    <span
      className="TabItem-name-0-1-9"
    >
      Sports & Outdoors
    </span>
  </div>
</div>
`;
